### ISSUE TITLE
Allow executing Butterfly passing a set of extension specific properties as input parameter

### ISSUE THREAD

**Issue by fabiocarvalho777** [paypal/butterfly#167]
### Use case

Some extensions would like the ability to reuse the same transformation template for different scenarios by configuring them with properties.

For example, a simple transformation template could perform text transformation by replacing values based on a regex. The new value to be applied could be configurable by using properties set as input parameter when running Butterfly.

The extension would be free to use those properties as they wish. In the prior example, they could also apply properties to the regular expression or files to be modified.

This way they don't have to duplicate the transformation template code for each of these scenarios.

### Passing the properties as input

* Programmatically: the Configuration object, which can be taken by the facade, should also allow setting a properties file.
* CLI: two new options should be introduced (if both are set, `q` is ignored and only `p` is used):
  * `-p` followed by a list of key value pairs separated by semi-colon (example `-pkey1=value1;key2=value2`)
  * `-q`The path to a properties file in the file system

### API

The properties should be made available to the application in two different ways:

- Transformation time: Butterfly should automatically fetch a property during transformation time whenever the transformation context attribute name starts with `$`.
  - If a TU `executeIf` check is used with a property name, then it must result in `false` whenever the property has not been set, is `null`, or different than the String value `true`.
- Definition time: **out of scope for this issue**. It will be taken care of in #309.

### Dependencies

This issue depends on #168
